signingscript/sign.py

```diff
@@ -350,31 +352,35 @@ def build_signtool_cmd(context, from_, fmt, to=None):
     # speed over disk space.
     tmp_dir = tempfile.mkdtemp(prefix="wvtar", dir=context.config['work_dir'])
     # Get file list
-    all_files = await _extract_tarfile(
-        context, orig_path, compression, tmp_dir=tmp_dir
-    )
+    all_files = await _get_tarfile_files(orig_path, compression)
     files_to_sign = _get_widevine_signing_files(all_files)
     log.debug("Widevine files to sign: {}".format(files_to_sign))
     if files_to_sign:
+        # Extract all files so we can create `precomplete` with the full
+        # file list
+        all_files = await _extract_tarfile(
+            context, orig_path, compression, tmp_dir=tmp_dir
+        )
         tasks = []
         # Sign the appropriate inner files
         for from_, fmt in files_to_sign.items():
+            from_ = os.path.join(tmp_dir, from_)
             # Move the sig location on mac. This should be noop on linux.
             to = _get_mac_sigpath(from_)
             log.debug("Adding {} to the sigfile paths...".format(to))
             makedirs(os.path.dirname(to))
             tasks.append(asyncio.ensure_future(sign_file(
                 context, from_, fmt, to=to
             )))
-            all_files.append("{}".format(to))
+            all_files.append(to)
         await raise_future_exceptions(tasks)
-        # Append sig_files to the archive
+        remove_extra_files(tmp_dir, all_files)
```

Comment:
Anything outside of all_files. When i looked at the logs, I saw lines like `copying path/to/file.sig.tmp to path/to/file.sig` and i was worried that without cleanup, we might have a lot of cruft in `precomplete`. This turns out to not be the case. However, now that I have the code, I think it's a good preventative measure.